File: update_projects.py
import os
import json

dirPath = 'projects_glbs'
outputPath = os.path.join(dirPath, 'projects.json')

if not os.path.exists(dirPath):
    print("Error: directory not found")
    exit(1)

files = os.listdir(dirPath)
glbFiles = [f for f in files if f.lower().endswith('.glb') and f.lower() != 'nyc_power_grid.geojson']

projectFiles = []
for f in glbFiles:
    readableName = f[:-4].upper()
    projectFiles.append({
        'name': readableName,
        'file': f"projects_glbs/{f}",
        'desc': f'Placeholder description for {readableName}. Structural metadata and environmental constraints will be populated here.',
        'link': '#'
    })

with open(outputPath, 'w') as f:
    json.dump(projectFiles, f, indent=2)

print(f"Successfully indexed {len(glbFiles)} project GLB files into {outputPath}!")

File: update_projects.js
import fs from 'fs';
import path from 'path';

const dirPath = path.join(process.cwd(), 'projects_glbs');
const outputPath = path.join(dirPath, 'projects.json');

// Read all files in the directory
fs.readdir(dirPath, (err, files) => {
  if (err) {
    console.error("Error reading projects_glbs directory:", err);
    process.exit(1);
  }

  // Filter out any non-glb objects or specific exceptions
  const glbFiles = files.filter(f => f.toLowerCase().endsWith('.glb') && f.toLowerCase() !== 'nyc_power_grid.geojson');

  // Build the array payload
  const projectFiles = glbFiles.map(file => {
    // Generate a readable name from the filename
    const readableName = file.replace(/\.glb$/i, '').toUpperCase();
    
    return {
      name: readableName,
      file: `projects_glbs/${file}`,
      desc: `Placeholder description for ${readableName}. Structural metadata and environmental constraints will be populated here.`,
      link: '#'
    };
  });

  // Write it to JSON
  fs.writeFile(outputPath, JSON.stringify(projectFiles, null, 2), (writeErr) => {
    if (writeErr) {
      console.error("Failed to write projects.json:", writeErr);
      process.exit(1);
    }
    console.log(`Successfully indexed ${glbFiles.length} project GLB files into projects_glbs/projects.json!`);
  });
});
